### ISSUE TITLE
Transaction rollback cause unhandled rejection

### ISSUE THREAD

**Issue by Niklv** [tgriesser/knex#3246]
Example code:

```js
const knex = require('knex')(/* config */);

process.on('unhandledRejection', err => {
    console.log('unhandled!');
    console.log(err);
});

async function test() {
    const trx = await knex.transaction();
    await trx.rollback();
    console.log('OK');
}

test();
```

Output:

```log
03:15:36.907 info: OK
03:15:36.915 error: Transaction rejected with non-error: undefined
03:15:36.915 info: unhandled!
03:15:36.917 info: Transaction rejected with non-error: undefined
```

I can not define error handler cause `.transaction()` returns `trx.initPromise`.
https://github.com/tgriesser/knex/blob/master/src/util/make-knex.js#L41-L52

My temporary workaround:
```js
const trx = await new Promise(resolve => 
    knex.transaction(trx => 
        resolve(trx)
    ).catch(() => {})
);
```

**Comment by kibertoad** [comment]
@Niklv See #3235 :D. You can try 0.17.1-next (the only version that included this change) and see if it causes any unwanted side-effects.
Had discussion with @elhigu recently about this. Quote:
```
There actually is kind of a good reason for forcing to have handler for rejects... transaction may be implicitly rejected from database side (actually even postgres does that by default if any of the queries done in transaction fails) and we should always have some kind of handler to catch those implicit rejects. That is actually something which for we should tests and documentation.
so basically one should always get result of transaction through that result promise that is given when transaction is created... i don't know... that stuff is pretty ugly 😒
we should try to come up some good syntax which handles explicit and implicit rejects nicely...
```

So current behaviour is more or less intended but could be (cautiously) improved.

**Comment by kibertoad** [comment]
And probably we need to expose same promise that you get from .transaction(callback) for .transaction() somehow.

**Comment by kibertoad** [comment]
@Niklv This issue was addressed in master. Would releasing 0.18.0-next3 with this change be helpful for you?

**Comment by Niklv** [comment]
@kibertoad Godd job, thank you! Will try it in my projects.

**Comment by kibertoad** [comment]
@Niklv All right, knex@0.18.0-next4 with this feature is out.
